### ISSUE TITLE
Implement and test new analytics feature

### ISSUE THREAD

**Issue by secretsauceai** [secretsauceai/precise-wakeword-model-maker#1]
The basic code for this is in `model_analytics.py`. 

**Issue**
The current reported analytics on the model is taken from the last epoch on training, this is however not always the chosen epoch (ie when using `precise-train` with `-sb`, the save best model parameter). 

In addition other metrics should be recorded such as:
* number of wake word and not wake word files for training
* the two types of 'best' models (epoch with the lowest loss, epoch with the lowest val_loss)

The selection of the models should be dependent on whether the loss or val_loss is being used to select the best epoch. 

- [x]  refactor `model_analytics.py` into a proper class used in `PreciseModelingOperations`
- [x] add report on number of files for wake and not-wake
- [x] test on model selection for several runs
